lighthouse-core/gather/computed/computed-artifact.js

```diff
@@ -38,18 +38,21 @@ class ComputedArtifact {
 
   /**
    * Request a computed artifact, caching the result on the input artifact.
-   * @param {!OBject} artifact
+   * @param {!Object} artifact
+   * @param {!Object=} artifacts
    * @return {!Promise}
    */
-  request(artifact) {
+  request(artifact, artifacts) {
```

Comment:
pretty interesting, but I like it. leads to more clean implementations.

can you rename to `allArtifacts` to clarify, since we have 3 diff sorts in here.